File: gpu_bartender/model_args.py
from dataclasses import dataclass, field

@dataclass
class ModelArgs:
    num_params: int = field(default=1)
    vocab_size: int = field(default=1)
    hidden_size: int = field(default=1)
    num_attention_heads: int = field(default=1)
    num_key_value_heads: int = field(default=1)
    intermediate_size: int = field(default=1)
    num_layers: int = field(default=1)

File: gpu_bartender/src/modelArgs.ts
export class ModelArgs {
    constructor(
        public numParams: number = 1,
        public vocabSize: number = 1,
        public hiddenSize: number = 1,
        public numAttentionHeads: number = 1,
        public numKeyValueHeads: number = 1,
        public intermediateSize: number = 1,
        public numLayers: number = 1
    ) {}
}
